### DISCUSSION TITLE
Shoutcast stream url?

### DISCUSSION THREAD

**Discussion by abirduphigh** [Vurv78/WebAudio#46] [Q&A]
In [WHITELIST.md](https://github.com/Vurv78/WebAudio/blob/main/WHITELIST.md), Shoutcast is shown as whitelisted by default (yp.shoutcast.com). I understand the URL for a shoutcast streaming file to be http://shoutcast-server-ip:port/stream.mp3, example: http://167.86.93.118:8000/stream.mp3. When I use that address in the E2 webAudio(s) function, I get the error that the URL isn't whitelisted. Any suggestions? Should I just use another streaming service to accomplish this?

Thanks!

**Comment by vurvdev** [comment]
It's meant to be used with urls like http://yp.shoutcast.com/sbin/tunein-station.pls?id=567807

If this does need a url different than the one above I'll have to look into it later to add it

**Reply by abirduphigh** [reply]
Thank you for clarifying. Unfortunately, I tried to call your example url with `WA = webAudio( "http://yp.shoutcast.com/sbin/tunein-station.pls?id=567807" )` and I not longer get the whitelist error but I do get this console error: 
```
[WA]: URL [http://yp.shoutcast.com/sbin/tunein-station.pls?id=567807] was incompatible for WebAudio; It is block-streamed!
```
I'm not entirely sure what that means or how to resolve.

I then tried to stream a different url by creating a **webaudio_whitelist.txt** file in my data folder and added the line: `https://wrfi.broadcasttool.stream` and tried to run `webAudio( "https://wrfi.broadcasttool.stream:80/stream" )` but I once again got the whitelist error. I'm sure the file is in the right directory because now http://yp.shoutcast.com is whitelisted too. Is there a certain syntax I need to use for **webaudio_whitelist.txt** to work correctly?

I'm sorry if this is already explained somewhere but I couldn't find any info on proper whitelisting in this repository.

**Reply by vurvdev** [reply]
> Thank you for clarifying. Unfortunately, I tried to call your example url with WA = webAudio( "http://yp.shoutcast.com/sbin/tunein-station.pls?id=567807" ) and I not longer get the whitelist error but I do get this console error:

Ah, the problem here is that garrysmod audio streams (IGmodAudioChannel) have multiple different types which inconsistently provide functionality like for pausing, setting playback time and whatnot.

Instead of just having them silently fail, I opted to just have it hard error. Should probably look into it if it's blocking so many streams, although not looking forward to reports of people not being able to call functions like xwa:setT`ime

Additionally I wanted to get this done alongside non-3D streams which would be significantly harder https://github.com/Vurv78/WebAudio/issues/11

> I then tried to stream a different url by creating a webaudio_whitelist.txt file in my data folder and added the line: https://wrfi.broadcasttool.stream and tried to run webAudio( "https://wrfi.broadcasttool.stream:80/stream" ) but I once again got the whitelist error.

The problem here is webAudio checks both the server and client whitelists. So if it isn't whitelisted on both (Or on just the server, if you do not have a custom whitelist on your client) it will fail. So add this entry instead to the server if you can

> I'm sure the file is in the right directory because now http://yp.shoutcast.com is whitelisted too. Is there a certain syntax I need to use for webaudio_whitelist.txt to work correctly?

It is in the format as seen here:
https://github.com/Vurv78/WebAudio/blob/2c5713d406ee50b085295b3211416acb4738e8a3/lua/autorun/webaudio.lua#L424-L430

**Reply by abirduphigh** [reply]
I really appreciate you helping me with this. I think I understand how to properly white list now.

Yet now I find, even without a whitelist text file located in either my server or my client, this url gives me a whitelist error, even though it is in the WHITELIST.md 
`https://translate.google.com/translate_tts?ie=UTF-8&tl=en&client=tw-ob&q=Hello%20World`

Sorry to drag this on. I feel like I'm just missing something obvious but I would love to get your addon running smoothly on my server!

Thanks,
Zac
